src/app/auth/components/singin/singin.component.ts

```diff
@@ -0,0 +1,35 @@
+import { Component } from '@angular/core';
+import { AuthService } from '../../services/auth.service';
+import { SignInBody } from '../../../core/types/auth.types';
+import { FormGroup, FormControl, Validators } from '@angular/forms';
+@Component({
+  selector: 'app-singin',
+  templateUrl: './singin.component.html',
+  styleUrls: ['./singin.component.scss'],
+})
+export class SinginComponent {
+  constructor(private auth: AuthService) {}
+  patterns = {
+    MIN_LENGTH: 8,
+    PATTERN_PASSWORD:
+      /^(?=.*?[A-Z])(?=.*?[a-z])(?=.*?[0-9])(?=.*?[#?!@$%^&*-]).{8,}$/,
+  };
+  authForm = new FormGroup({
+    login: new FormControl('', [Validators.required, Validators.email]),
+    password: new FormControl('', [
+      Validators.required,
+      Validators.minLength(this.patterns.MIN_LENGTH),
+      Validators.pattern(this.patterns.PATTERN_PASSWORD),
+    ]),
+  });
+  controlLogin = this.authForm.get('login') as FormControl;
+  controlPassword = this.authForm.get('password') as FormControl;
+  onSingInButton(): void {
+    const data = this.authForm.value as SignInBody;
+    localStorage.setItem('login', data.login);
```

Comment:
I would recommend moving logic related to localStorage to separate service. In case, the project migrate to some other storage you cat fix all necessary staff in one place